### PR TITLE
[test gap] Add a new test case to verify lldp entry after restart syncd and orchagent

### PR DESCRIPTION
**Pull request by ZhaohuiS**
…cd and orchagent

### Description of PR

Summary:
Fixes # (issue)
Address https://github.com/sonic-net/sonic-mgmt/issues/15889

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
Add a new test case to address https://github.com/sonic-net/sonic-mgmt/issues/15889

#### How did you do it?
Restart syncd and orchagent, then check lldp entries in APPL_DB

#### How did you verify/test it?
run test_lldp_syncd.py

### CODE REVIEW

**Comment by ZhaohuiS** [comment]
Hold on the merge until https://github.com/sonic-net/sonic-buildimage/pull/21065 is merged, which is the image fix.

**Comment by ZhaohuiS** [comment]
@StormLiangMS Since image fix was merged in https://github.com/sonic-net/sonic-buildimage/pull/21065, could you please review and approve this test gap PR?

File: tests/lldp/test_lldp_syncd.py [hunk 1]
```diff
@@ -250,7 +250,56 @@ def test_lldp_entry_table_content(
...
+    # To get lldp entry keys again after all interfaces are up
+    lldp_entry_keys = get_lldp_entry_keys(db_instance)
+    lldpctl_output = get_lldpctl_output(duthost)
+    show_lldp_table_int_list = get_show_lldp_table_output(duthost)
+    lldpctl_interfaces = lldpctl_output["lldp"]["interface"]
+    assert_lldp_interfaces(
+        lldp_entry_keys, show_lldp_table_int_list, lldpctl_interfaces
+    )
+    for interface in get_lldp_entry_keys(db_instance):
+        entry_content = get_lldp_entry_content(db_instance, interface)
```
Comment (StormLiangMS) [inline tests/lldp/test_lldp_syncd.py:290]: This seems repeating code in this file, could we move them into a function? Too much code and logic inside this test to make it hard to read through, and distracting from the main function.

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Comment by cyw233** [comment]
Hey @ZhaohuiS, have we confirmed this change is compatible with multi-asic devices (e.g. T2), please? From my observation, the `docker restart syncd` or `docker restart swss` command might have issues on multi-asic devices because these devices are having containers like `syncd0, syncd1, swss0, swss1...`

**Comment by ZhaohuiS** [comment]
> Hey @ZhaohuiS, have we confirmed this change is compatible with multi-asic devices (e.g. T2), please? From my observation, the `docker restart syncd` or `docker restart swss` command might have issues on multi-asic devices because these devices are having containers like `syncd0, syncd1, swss0, swss1...`

Sorry, no, I don't have resource to verify it on T2, if there is some issue on T2, please help fix it.

**Comment by cyw233** [comment]
> > Hey @ZhaohuiS, have we confirmed this change is compatible with multi-asic devices (e.g. T2), please? From my observation, the `docker restart syncd` or `docker restart swss` command might have issues on multi-asic devices because these devices are having containers like `syncd0, syncd1, swss0, swss1...`
> 
> Sorry, no, I don't have resource to verify it on T2, if there is some issue on T2, please help fix it.

No worries, Microsoft ADO 30648046 created for this.

**Pull request merged**